### PR TITLE
fix(aad): fix aad to support multi env input

### PR DESCRIPTION
**Pull request by KennethBWSong**
task: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_sprints/taskboard/AuthAndData/Microsoft%20Teams%20Extensibility/CY21-10.2?workitem=12382197

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/aad/utils/common.ts [hunk 1]
```diff
@@ -63,4 +64,30 @@ export class Utils {
...
+      return skip;
+    }
+
+    const objectId = ctx.envInfo.config.auth?.objectId;
+    const clientId = ctx.envInfo.config.auth?.clientId;
+    const oauth2PermissionScopeId = ctx.envInfo.config.auth?.accessAsUserScopeId;
+    const clientSecret = ctx.envInfo.config.auth?.clientSecret;
+    const skip = ctx.envInfo.config.auth?.[ConfigKeys.skip];
+
+    if (objectId && clientId && oauth2PermissionScopeId && clientSecret) {
```
Comment (SLdragon) [inline packages/fx-core/src/plugins/resource/aad/utils/common.ts:80]: Do we need to throw errors when something is undefined?

Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/aad/utils/common.ts:80]: Now check will be performed separately when provisioning or skipping provision. It's a good idea to throw error here. I will update the code. Thx!

Comment (adashen) [reply packages/fx-core/src/plugins/resource/aad/utils/common.ts:80]: Failure one information not complete?

File: packages/fx-core/src/plugins/resource/aad/utils/common.ts [hunk 2]
```diff
@@ -63,4 +64,30 @@ export class Utils {
...
+    if (!isMultiEnvEnabled()) {
+      const skip = ctx.config.get(ConfigKeys.skip) as boolean;
+      return skip;
+    }
+
+    const objectId = ctx.envInfo.config.auth?.objectId;
+    const clientId = ctx.envInfo.config.auth?.clientId;
+    const oauth2PermissionScopeId = ctx.envInfo.config.auth?.accessAsUserScopeId;
+    const clientSecret = ctx.envInfo.config.auth?.clientSecret;
+    const skip = ctx.envInfo.config.auth?.[ConfigKeys.skip];
```
Comment (dooriya) [inline packages/fx-core/src/plugins/resource/aad/utils/common.ts:78]: According to the env config schema: https://aka.ms/teamsfx-env-config-schema, we don't have the `skip` config key in the new env config.

Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/aad/utils/common.ts:78]: I will update the schema. Thx!

File: packages/fx-core/src/plugins/resource/aad/utils/common.ts [hunk 3]
```diff
@@ -63,4 +64,30 @@ export class Utils {
     const tenantId: string = (tokenObject as any).tid;
     return tenantId;
   }
+
+  public static getAndMoveInput(ctx: PluginContext): boolean {
```
Comment (dooriya) [inline packages/fx-core/src/plugins/resource/aad/utils/common.ts:68]: I cannot guess what this function is used for without go through the code? Can we provide a better naming for it?

File: packages/fx-core/src/plugins/resource/aad/utils/common.ts (hunk 2 above)
Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/aad/utils/common.ts:80]: Will fix in another pr, thx!

Comment (dooriya) [reply packages/fx-core/src/plugins/resource/aad/utils/common.ts:80]: OK. So can you remove this line which is reading `skip` property from env config? You can consolidate the skip aad provision/configuration logic in another PR.

**Pull request merged**